applicationset/webhook/webhook.go

```diff
@@ -167,6 +170,8 @@ func (h *WebhookHandler) Handler(w http.ResponseWriter, r *http.Request) {
 		payload, err = h.gitlab.Parse(r, gitlab.PushEvents, gitlab.TagEvents, gitlab.MergeRequestEvents)
 	case r.Header.Get("X-Vss-Activityid") != "":
 		payload, err = h.azuredevops.Parse(r, azuredevops.GitPushEventType, azuredevops.GitPullRequestCreatedEventType, azuredevops.GitPullRequestUpdatedEventType, azuredevops.GitPullRequestMergedEventType)
+	case r.Header.Get("X-Custom-Event") != "":
+		payload, err = parseCustomPayload(r)
```

Comment:
Is there a specific reason to define a custom header with a specific payload?

Since GitLab and GitHub have their own payload structures, I make sure to differentiate between the content they send and the internal part of ArgoCD that we are responsible for handling.

This is why I initially suggested the query string filter interface it seemed to me the most agnostic solution.

Reply:
My first intend was to use query string filter as you mention, but I was not sure how to handle this correctly and be able to differentiate between payload structures.

Reading code I saw lines like :
```
  if r.Method != http.MethodPost {
     status = http.StatusMethodNotAllowed
  } 

  req := httptest.NewRequest(http.MethodPost, "/api/webhook", nil)
```

Not an expert on REST API, but generally query string imply GET method and when you use a POST method, parameters are inside body request

That's why I switch to use custom header+payload, and to be consistent with other cases.

That being said, I think query string are tolerated in POST method.
And at end, what we can do is simply :
```
	default:
		ls := r.URL.Query().Get("labelsSelector")
		if ls != "" {
		    // Parse the labelsSelector query from the URL query parameter and continue
		} else {
			log.Debug("Ignoring unknown webhook event")
			http.Error(w, "Unknown webhook event", http.StatusBadRequest)
			return
		}
```

What is sure is that create a custom header/payload, should be documented and maintained.
Don't know what is the best ?

Reply:
Hello,

Thanks again for your feedback and those insights :+1: 
I've the feeling that this PR is aligned on most of the ideas. 

I think I will have more time to work on this in the coming weeks.

@crenshaw-dev if you have any idea/thought, please feel free to comment :pray: